### PR TITLE
Add Cluster status docs

### PR DESCRIPTION
**Pull request by cxbrowne1207**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/clustermgmt/cluster-status.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+First set the `CLUSTER_NAME` and `KUBECONFIG` environment variables.
+
+```
+export CLUSTER_NAME=w01
+export KUBECONFIG=${CLUSTER_NAME}/${CLUSTER_NAME}-eks-a-cluster.kubeconfig
+```
+
+To view an EKS Anywhere cluster status, use `kubectl` command show the status of the cluster object.
+The output should show the `yaml` definition of the EKS Anywhere `Cluster` object which has a `status` field
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-status.md:24]:
```suggestion
The output should show the `yaml` definition of the EKS Anywhere `Cluster` object which has a `status` field.
```

File: docs/content/en/docs/clustermgmt/cluster-status.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+```
+kubectl get clusters $CLUSTER_NAME -o yaml
+...
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-status.md:28]: I think it's good to keep the output here.

File: docs/content/en/docs/clustermgmt/cluster-status.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Machine readable value about a terminal problem while reconciling the cluster set at the same time as the `status.failureMessage`.
+
+**`status.conditions`**
+
+Provides a collection of condition objects that report a high-level assessment of cluster readiness.
+
+**`status.observedGeneration`**
+
+This is the latest generation observed, set everytime the status is updated. If this is not the same as the cluster object's `metadata.generation`, it means that the status being view represents an old generation of the cluster specification and is not up-to-date yet.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-status.md:67]:
```suggestion
This is the latest generation observed, set everytime the status is updated. If this is not the same as the cluster object's `metadata.generation`, it means that the status being viewed represents an old generation of the cluster specification and is not up-to-date yet. 
```

File: docs/content/en/docs/clustermgmt/cluster-status.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+This is the latest generation observed, set everytime the status is updated. If this is not the same as the cluster object's `metadata.generation`, it means that the status being view represents an old generation of the cluster specification and is not up-to-date yet. 
+
+### Cluster status conditions
+
+Conditions provide a high-level status report representing an assessment of cluster readiness using a collection of conditions each of a particular type. The following condition types are supported:
+
+  * `ControlPlaneInitialized` - reports the first control plane has been initialized and the cluster is contactable with the kubeconfig. Once this condition is marked `True`, its value never changes.
+
+  * `ControlPlaneReady` -  reports that the condition of the current state of the control plane with respect to the desired state specified in the Cluster specification. This condition is marked `True` once the number of control plane nodes in the cluster match the expected number of worker nodes as defined in the cluster specifications and all the control plane nodes are up to date and ready.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-status.md:75]:
```suggestion
  * `ControlPlaneReady` -  reports that the condition of the current state of the control plane with respect to the desired state specified in the Cluster specification. This condition is marked `True` once the number of control plane nodes in the cluster match the expected number of control plane nodes as defined in the cluster specifications and all the control plane nodes are up to date and ready.
```
Is this what you mean?

File: docs/content/en/docs/clustermgmt/cluster-status.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+  * `DefaultCNIConfigured` - reports the configuration state of the default CNI specified in the cluster specifications. It will be marked as `True` once the default CNI has been successfully configured on the cluster. 
+  However, if the EKS Anywhere default cilium CNI has been [configured to skip upgrades]({{< relref "../getting-started/optional/cni/#use-a-custom-cni" >}}) in cluster specification, then this condition will be marked as `False` with the reason `SkipUpgradesForDefaultCNIConfigured`.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-status.md:78]:
```suggestion
  However, if the EKS Anywhere default cilium CNI has been [configured to skip upgrades]({{< relref "../getting-started/optional/cni/#use-a-custom-cni" >}}) in the cluster specification, then this condition will be marked as `False` with the reason `SkipUpgradesForDefaultCNIConfigured`.
```

File: docs/content/en/docs/clustermgmt/cluster-status.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+  * `WorkersReady` - reports that the condition of the current state of worker machines vs the desired state specified in the Cluster specification. This condition is marked `True` once the number of worker nodes in the cluster match the expected number of worker nodes as defined in the cluster specifications and all the worker plane nodes are up to date and ready.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-status.md:80]:
```suggestion
  * `WorkersReady` - reports that the condition of the current state of worker machines versus the desired state specified in the Cluster specification. This condition is marked `True` once the number of worker nodes in the cluster match the expected number of worker nodes as defined in the cluster specifications and all the worker plane nodes are up to date and ready.
```

File: docs/content/en/docs/clustermgmt/cluster-terraform.md [hunk 7]
```diff
@@ -166,7 +166,38 @@ Follow these steps if you want to use your initial cluster to create and manage
...
 
-6. The kubeconfig for your new cluster is stored as a secret on the management cluster.
+6. Check the state of a cluster using `kubectl` to show the cluster object with its status.
+   
+   The `status` field on the cluster object field holds information about the current state of the cluster.
+
+   ```
+   kubectl get clusters w01 -o yaml
+   ...
+   status:
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-terraform.md:176]: See comment above about deleting output here.

File: docs/content/en/docs/clustermgmt/cluster-upgrades/baremetal-upgrades.md [hunk 8]
```diff
@@ -177,6 +177,37 @@ and then you will run the [upgrade cluster command]({{< relref "baremetal-upgrad
...
   --kubeconfig mgmt/mgmt-eks-a-cluster.kubeconfig
    ```
+ 
+    To check the state of a cluster managed with the cluster lifecyle feature, use `kubectl` to show the cluster object with its status.
+    
+    The `status` field on the cluster object field holds information about the current state of the cluster.
+
+    ```
+    kubectl get clusters w01 -o yaml
+    ...
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/baremetal-upgrades.md:187]: See comment above about deleting output here.

File: docs/content/en/docs/getting-started/baremetal/baremetal-getstarted.md [hunk 9]
```diff
@@ -184,6 +184,38 @@ Follow these steps if you want to use your initial cluster to create and manage
...
       ```
+
+      To check the state of a cluster managed with the cluster lifecyle feature, use `kubectl` to show the cluster object with its status.
+      
+      The `status` field on the cluster object field holds information about the current state of the cluster.
+
+      ```
+      kubectl get clusters w01 -o yaml
+      ...
+      status:
```
Comment (chrisnegus) [inline docs/content/en/docs/getting-started/baremetal/baremetal-getstarted.md:195]: See comment above about deleting output here.

File: docs/content/en/docs/getting-started/cloudstack/cloudstack-getstarted.md [hunk 10]
```diff
@@ -233,6 +233,38 @@ Follow these steps if you want to use your initial cluster to create and manage
...
       kubectl apply -f eksa-w01-cluster.yaml 
       ```
+
+       To check the state of a cluster managed with the cluster lifecyle feature, use `kubectl` to show the cluster object with its status.
+      
+      The `status` field on the cluster object field holds information about the current state of the cluster.
+
+      ```
+      kubectl get clusters w01 -o yaml
+      ...
```
Comment (chrisnegus) [inline docs/content/en/docs/getting-started/cloudstack/cloudstack-getstarted.md:243]: See comment above about deleting output here.

File: docs/content/en/docs/getting-started/snow/snow-getstarted.md [hunk 11]
```diff
@@ -209,6 +209,38 @@ Follow these steps if you want to use your initial cluster to create and manage
...
       ```
+
+       To check the state of a cluster managed with the cluster lifecyle feature, use `kubectl` to show the cluster object with its status.
+      
+      The `status` field on the cluster object field holds information about the current state of the cluster.
+
+      ```
+      kubectl get clusters w01 -o yaml
+      ...
+      status:
```
Comment (chrisnegus) [inline docs/content/en/docs/getting-started/snow/snow-getstarted.md:220]: See comment above about deleting output here.

File: docs/content/en/docs/getting-started/vsphere/vsphere-getstarted.md [hunk 12]
```diff
@@ -216,6 +216,38 @@ Follow these steps if you want to use your initial cluster to create and manage
       kubectl apply -f eksa-w01-cluster.yaml 
       ```
 
+       To check the state of a cluster managed with the cluster lifecyle feature, use `kubectl` to show the cluster object with its status.
+      
+      The `status` field on the cluster object field holds information about the current state of the cluster.
+
+      ```
+      kubectl get clusters w01 -o yaml
+      ...
```
Comment (chrisnegus) [inline docs/content/en/docs/getting-started/vsphere/vsphere-getstarted.md:225]: See comment above about deleting output here.

**Review by chrisnegus — COMMENTED** [review]
Looks good. I made some suggestions about not duplicating output across all the pages, just to tighten it up a bit.

File: docs/content/en/docs/clustermgmt/cluster-status.md (hunk 4 above)
Comment (cxbrowne1207) [reply docs/content/en/docs/clustermgmt/cluster-status.md:75]: yes, thanks for the catch

**Review by chrisnegus — APPROVED** [review]
/lgtm

**Pull request merged**